File: online-judge-solutions/Codeforces/1744B.py
#https://codeforces.com/problemset/problem/1744/B
#Lang		:	Python 3.8
#Time		:	1029 ms
#Memory         :	12400 KB
for _ in range(int(input())):
    n,q=map(int,input().split())
    s=impar=par=0
    a=list(map(int,input().split()))[:n]
    s,m=sum(a),{}
    for i in a:
        if(i%2==0):par+=1
        else:impar+=1
    m[0],m[1]=par,impar
    for i in range(q):
        typ,x=map(int,input().split())
        s+=m[typ]*x
        if(m[typ]>0 and x%2==1):
            m[1-typ],m[typ]=n,0
        print(s)

File: online-judge-solutions/Codeforces/1744B.cpp
//https://codeforces.com/problemset/problem/1744/B
//Lang		:	GNU C++14
//Time		:	374 ms
//Memory	:	1000 KB
#include<iostream>
#include<map>
using namespace std;
#define fastInp cin.tie(0); cout.tie(0); ios_base::sync_with_stdio(0);
int main(){
	fastInp;
	long long t,n,q,a;
	for(cin>>t;t--;){
		cin>>n>>q;
		long long type,x,s=0,impar=0,par=0;
		for(int i=0;i<n;i++){
			cin>>a;
			s+=a;
			(a%2==0)?par++:impar++;
		}
		map<int,int>m;
		m[0]=par,m[1]=impar;
		for(int i=0;i<q;i++){
			cin>>type>>x;
			s+=m[type]*x;
			if(m[type]>0 && x%2==1){m[1-type]=n;m[type]=0;}
			cout<<s<<endl;
		}
	}
	return 0;
}
